### PR TITLE
Add: aarch64/arm64 and ppc64le

### PR DESCRIPTION
**Pull request by ax3l**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

**Comment by ax3l** [comment]
OSX ARM64 w/ OpenMPI is the only test acting up:
```
-- Checking for module 'mpi-cxx'
--   No package 'mpi-cxx' found
CMake Error at /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find MPI (missing: MPI_CXX_FOUND)
Call Stack (most recent call first):
  /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindMPI.cmake:1841 (find_package_handle_standard_args)
  CMakeLists.txt:96 (find_package)


-- Could NOT find MPI_CXX (missing: MPI_CXX_LIB_NAMES MPI_CXX_WORKS) 
```

**Comment by ax3l** [comment]
Maybe @jakirkham @isuruf know this:

OSX ARM64 w/ OpenMPI is the only test acting up:
```
-- Checking for module 'mpi-cxx'
--   No package 'mpi-cxx' found
CMake Error at /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find MPI (missing: MPI_CXX_FOUND)
Call Stack (most recent call first):
  /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /Users/runner/miniforge3/conda-bld/heffte_1725317125912/_build_env/share/cmake-3.30/Modules/FindMPI.cmake:1841 (find_package_handle_standard_args)
  CMakeLists.txt:96 (find_package)


-- Could NOT find MPI_CXX (missing: MPI_CXX_LIB_NAMES MPI_CXX_WORKS) 
```

**Comment by ax3l** [comment]
@conda-forge-admin, please rerender

File: recipe/run_test.sh [hunk 1]
```diff
@@ -7,14 +7,20 @@ export OMP_NUM_THREADS=2
 
 # OpenMPI
 export OMPI_MCA_plm=^rsh
-# note: example_r2r uses 4 MPI ranks, CI has 2 cores
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" && "${mpi}" == "openmpi" ]]; then
```
Comment (ax3l) [inline recipe/run_test.sh]: @isuruf Can I set `CONDA_BUILD_CROSS_COMPILATION` or an equivalent in `run_tests.sh` as well? It appears to be undefined here.

File: recipe/run_test.sh [hunk 2]
```diff
@@ -7,14 +7,20 @@ export OMP_NUM_THREADS=2
...
+  export OPAL_PREFIX="$PREFIX"
+fi
```
Comment (ax3l) [inline recipe/run_test.sh]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -39,6 +39,7 @@ requirements:
     - {{ stdlib("c") }}
     - cmake
     - make  # [unix]
+    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]
```
Comment (ax3l) [inline recipe/meta.yaml]:
```suggestion
    # In OpenMPI, the compiler wrappers are binaries and the wrappers in build
    # can use host libraries by adding OPAL_PREFIX and in mpich, compiler
    # wrappers are bash scripts and wrappers in build can't use host libraries.
    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -39,7 +39,10 @@ requirements:
     - {{ stdlib("c") }}
     - cmake
     - make  # [unix]
-    - pkg-config
```
Comment (jakirkham) [inline recipe/meta.yaml:42]: Do we still need `pkg-config` for the build?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -39,7 +39,10 @@ requirements:
...
+    # In OpenMPI, the compiler wrappers are binaries and the wrappers in build
+    # can use host libraries by adding OPAL_PREFIX and in mpich, compiler
+    # wrappers are bash scripts and wrappers in build can't use host libraries.
+    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]    - pkg-config
```
Comment (jakirkham) [inline recipe/meta.yaml:45]: Looks like this snuck in

```suggestion
    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -55,6 +58,10 @@ test:
     - cmake
     - make  # [unix]
     - pkg-config
+    # In OpenMPI, the compiler wrappers are binaries and the wrappers in build
+    # can use host libraries by adding OPAL_PREFIX and in mpich, compiler
+    # wrappers are bash scripts and wrappers in build can't use host libraries.
+    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]    - pkg-config
```
Comment (jakirkham) [inline recipe/meta.yaml:64]: Same here

```suggestion
    - {{ mpi }}  # [not win and build_platform != target_platform and mpi == "openmpi"]
```

File: recipe/run_test.sh [hunk 7]
```diff
@@ -7,14 +7,17 @@ export OMP_NUM_THREADS=2
 
 # OpenMPI
 export OMPI_MCA_plm=^rsh
-# note: example_r2r uses 4 MPI ranks, CI has 2 cores
```
Comment (jakirkham) [inline recipe/run_test.sh:10]: Should we move this down by the relevant test as well? Or does it still need to be here?

**Review by jakirkham — COMMENTED** [review]
Thanks Axel! 🙏

Sorry was out when this came up. Glad you were able to work through things here

Asked a couple questions below

**Comment by ax3l** [comment]
Thank you, @jakirkham 😊 

I have a follow up open in #7 where we can include your suggestions, looks good to me!

File: recipe/meta.yaml (hunk 4 above)
Comment (ax3l) [reply recipe/meta.yaml:42]: I think not here, I use this sometimes for FFTW but in the end this seems leftover

**Pull request merged**